src/formatDate.js

```diff
@@ -48,9 +48,51 @@
  *
  * @returns {string}
  */
+const CENTURY_INDICATOR = 30;
+const CENTURY_21 = '20';
+const CENTURY_20 = '19';
+const YEAR_FORMAT_OF_4 = 'YYYY';
+const YEAR_FORMAT_OF_2 = 'YY';
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateArray = date.split(fromSeparator);
+  const newDateArray = [];
+
+  formatYear(fromFormat, toFormat, dateArray);
+
+  for (let i = 0; i < dateArray.length; i++) {
+    newDateArray[i] = dateArray[fromFormat.indexOf(toFormat[i])];
```

Comment:
Now everything is written correctly, but it's a bit hard to read, you could move the search for the new index into a function:
```js
function findFormatIndex(oldFormat, newFormat, i) {
  return oldFormat.indexOf(newFormat[i]);
}
```